### PR TITLE
Fixed Reading List on Mobile + Small Improvements

### LINKED ISSUE #11983: Reading list view broken in mobile
**Describe the bug**

Reading list view is not responsive in mobile

**To Reproduce**

1. Go to https://dev.to/readinglist
2. Reading list layout wont be responsive 

**Expected behavior**
 Readinglist should be responsive in mobile

**Screenshots**

Screenshot_20201220-132914_Brave

**Smartphone**

- Device: readmi note 6 pro
- OS, version: 10
- Browser, version: 1.17.75

### PR DESCRIPTION
**Pull request by nickytonline**
<details>
<summary>Details</summary>

</details>

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Implements changes proposed in #11983. One notable change is that [we only filter by one tag at a time now](https://github.com/forem/forem/issues/11033#issuecomment-717087477).

Before

https://user-images.githubusercontent.com/833231/110536113-14d56300-80ef-11eb-9586-5b085c17b0c2.mp4

After

https://user-images.githubusercontent.com/833231/113223886-2a562c80-9258-11eb-9caf-b6497f2073ef.mp4

## Related Tickets & Documents

Closes #11983

## QA Instructions, Screenshots, Recordings

The reading list section should still be able to archive and unarchive reading list items.

### Archiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
2. Click on the _Archive_ button of the reading list item.
4. The archived item will disappear from the reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.

### Unarchiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
3. Click on the _Archive_ button of the reading list item.
4. The unarchived item will disappear from the archived reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.
6. Navigate to http://localhost:3000/readinglist/archive
7. Click on the _Unarchive_ button of the archived reading list item.
8. A snackbar item at the bottom left of the screen should say "Unarchiving..." and then disappear after a second or two.

#### Filtering the Reading List

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list. Do this with 4 or 5 articles so the reading list is a small size.
3. Navigate to http://localhost:3000/readinglist
4. In the search text box, enter in some text that is in the title of an article. It will filter so only article titles with that text are visible.
5. In the filter by tag drop down, select a tag. It will filter so only article titles with the current filter text and articles with the selected tag are visible.
6. Clear the search text box. It will filter so only articles with the selected tag are visible.

#### Filtering the Archive

To test the same functionality in the archived view:

1. From the reading list with the existing list from the previous test set up, click on the archive button of an item in the reading list for as many articles as you'd like.
2. Repeat steps 4-6 from the "Filtering the Reading List" section of QA.

### UI accessibility concerns?

I fixed the markup so that everything is contained within the `<main />` landmark now. There is a colour contrast issue with the `<select />` being reported, but this is not related to the work in the PR. This appears to be more of a global issue with `<select />`s. cc: @ludwiczakpawel 

@aitchiss, the fieldset on the left when on the medium breakpoint and up is now a group of radio buttons since we only allow for filtering on one tag now. The keyboard navigation appears to be working well as is Voice Over, but please take it for a spin to see if there are any gaps in the work there.

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

RHCP on the Simpsons where Flea says "Wow! That's much better. Everyone can enjoy that."

### CODE REVIEW

**Comment by nickytonline** [comment]
I've scrapped some component tests that existed for this and will write some end to end tests instead.

File: app/javascript/readingList/components/TagList.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        onSelectTag(event);
+      }}
+      onChange={onSelectTag}
+    >
+      {selectedTag === '' && <option>Select a tag...</option>}
+      {availableTags.map((tag) => (
+        <option
+          selected={tag === selectedTag}
+          className={`crayons-link crayons-link--block ${
+            tag === selectedTag ? 'crayons-link--current' : ''
```
Comment (ludwiczakpawel) [inline app/javascript/readingList/components/TagList.jsx:34]: Out of curiosity, why not this:

```
tag === selectedTag && 'crayons-link--current'
```

^ since there's no alternative value anyway..?

File: app/javascript/readingList/readingList.jsx [hunk 2]
```diff
@@ -13,32 +12,49 @@ import {
 import { ItemListItem } from './components/ItemListItem';
 import { ItemListItemArchiveButton } from './components/ItemListItemArchiveButton';
```
Comment (ludwiczakpawel) [inline app/javascript/readingList/readingList.jsx:13]: We should at some point look into components names: `ItemListItem` is probably not very descriptive...

File: app/javascript/readingList/components/TagList.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        // We need blur for a11y, but we also don't want to make the same search query twice
+        // if the tag hasn't changed since the previous onChange.
+        const { value } = event.target;
+
+        if (value === selectedTag) {
+          return;
+        }
+
+        onSelectTag(event);
+      }}
```
Comment (ludwiczakpawel) [inline app/javascript/readingList/components/TagList.jsx:26]: Can you help me understand what does it do? I couldn't notice any difference when I removed this part of code.

**Review by ludwiczakpawel — CHANGES_REQUESTED** [review]
I know it's a Draft PR but you added me as reviewers so here I am :D

1. Desktop UI regression. This one is IMO a blocker - afair we didn't want to replace sidebar nav with `select` field everywhere but only on smaller breakpoint(s). Bigger breakpoints should keep the left sidebar. This is similar to what we do on notifications and settings and that's likely the pattern we want to keep elsewhere.
    image

2. Going back to main view - blocker as well. When I choose a tag I can no longer go back to "main" view of Reading list - without any tag selected. That should probably be its own option in the select like "Everything" or something..
3. a11y issue - blocker too. When I use keyboard to choose a tag in this select, then hit `Shift + Tab` to go to previous field (search field), the filtering is being reset:
    
    https://user-images.githubusercontent.com/108287/108681811-e2711680-74ef-11eb-91d5-a8278d10c9a8.mov

    It's probably because search field, even if empty, is running a call and returns results matching the query (query is empty so it return everything). In other words, we should probably limit search field to only search through current view (selectedTag).

---

Not a blocker but something to think about in future is being able to reflect the current filter with URL. In other words, when you change the filtering we could update the browser's URL too to `dev.to/readinglist/javascript`. I'm aware it's something we don't have right now so there's no expectation to build it :D But you know, just something to keep in back of our heads.

**Comment by nickytonline** [comment]
Ahh, I must have misread the comment about select. I thought we wanted it for desktop as well.

I'll look at the other issues as well you mentioned.

In regards to keeping the filter in the URL, I completely agree, but that should be a separate PR a that changes how the entire state is managed for this component.

**Comment by ludwiczakpawel** [comment]
@nickytonline just to make sure I'm not making things up right now :D: 

> **Expected behavior**
> ...
> 2. Collapse nav into the select on mobile breakpoint.

^ that's also being mentioned in other comments in the same [issue](https://github.com/forem/forem/issues/11033).

File: app/javascript/readingList/components/TagList.jsx (hunk 1 above)
Comment (nickytonline) [reply app/javascript/readingList/components/TagList.jsx:34]: If you do it this way, it will add a CSS class name called "false".

Comment (ludwiczakpawel) [reply app/javascript/readingList/components/TagList.jsx:34]: 😲 javascript is fun...

**Comment by nickytonline** [comment]
@ludwiczakpawel, @lisasy I've reworked this. The sidebar remains until we hit the smallest breakpoint. I took some liberties from some minor spacing changes, so let me know what you think.

@aitchiss, I also reworked the side bar to be a group of checkboxes even though visually it looks the same. It helps with the keyboard nav and voice over.

Here is the loom video, https://www.loom.com/share/c78e3427cfdf4b5ba01506ddd01ace32

Tests are still broken, but I will sort that out.

**Comment by nickytonline** [comment]
I'm putting this on hold after I'm back as I'm looking into a higher priority issue in tegards to service workers, but if you can take this for a test drive @lisasy, @ludwiczakpawel, and @aitchiss, it'd be much appreciated. The only blocker for this PR at the moment is tests.

**Comment by nickytonline** [comment]
As mentioned, I'll resume this when I'm back from my time off, but I put up some e2e tests @aitchiss. I discovered that Cypress is not evaluating `window.matchMedia` properly and I think it's related to https://github.com/cypress-io/cypress/issues/311. The browser API is working, but it always returns false when trying to match breakpoints.

**Comment by ludwiczakpawel** [comment]
@nickytonline just turn it to "Ready for review" when ready :)

**Comment by lisasy** [comment]
I can't seem to review this PR on my local dev, so my feedback is based on the video in the description. On the smallest breakpoint, there ought to be appropriate margin around the heading, search input, dropdown, and button so that these components aren't touching the edges of the device.

**Comment by nickytonline** [comment]
> I can't seem to review this PR on my local dev, so my feedback is based on the video in the description. On the smallest breakpoint, there ought to be appropriate margin around the heading, search input, dropdown, and button so that these components aren't touching the edges of the device.

That seems to be the way the crayon layout class works, but I can definitely add margin. Also, those videos are out of date as that was my initial pass. I'll be posting updated videos shortly and will pull this out of draft mode.

File: app/javascript/readingList/readingList.jsx [hunk 4]
```diff
@@ -73,11 +91,11 @@ export class ReadingList extends Component {
       : READING_LIST_PATH;
 
     // empty items so that changing the view will start from scratch
-    this.setState({ statusView: newStatusView, items: [] });
+    this.setState({ statusView: newStatusView, items: [], selectedTag });
 
     this.search(query, {
       page: 0,
-      tags: selectedTags,
+      tags: [],
```
Comment (nickytonline) [inline app/javascript/readingList/readingList.jsx:98]: Although we're filtering by one tag now, the search query still requires an array which is why this gets reset to an empty array.

File: app/javascript/readingList/readingList.jsx [hunk 5]
```diff
@@ -210,30 +217,65 @@ export class ReadingList extends Component {
...
-          </main>
-
-          {snackBar}
-        </div>
-      </section>
+          </fieldset>
+        </header>
+        <MediaQuery
+          query={`(width >= ${BREAKPOINTS.Medium}px)`}
+          render={(matches) => {
```
Comment (nickytonline) [inline app/javascript/readingList/readingList.jsx:239]: @aitchiss as [per your question in the PR for creating the `<MediaQuery />`](https://github.com/forem/forem/pull/12809#discussion_r582669298) component, this is why I opted for a render prop.

File: cypress/fixtures/search/readingList.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
+{
```
Comment (nickytonline) [inline cypress/fixtures/search/readingList.json:1]: Reading list mocked data for e2e tests.

File: jest.config.js [hunk 7]
```diff
@@ -25,8 +25,8 @@ module.exports = {
   ],
   coverageThreshold: {
     global: {
-      statements: 43,
-      branches: 39,
+      statements: 42,
```
Comment (nickytonline) [inline jest.config.js:28]: I reduced the coverage thresholds because the component tests for the reading list no longer exist. I've converted them to E2E tests. @djuber, this is something I haven't had time to look into/do yet, but it'd be good to aggregate the test coverage from jest with that of Cypress so that we can get a true picture of the code coverage for the front-end.

**Comment by nickytonline** [comment]
There are some flashes of content that I'm going to look into which existed previously. It might merit a separate PR though.


https://user-images.githubusercontent.com/833231/110524551-8ad2cd80-80e1-11eb-9b13-8ed1829f19b1.mp4

**Comment by nickytonline** [comment]
Just sorting out some E2E test failures...

**Review by aitchiss — COMMENTED** [review]
In Chrome I'm not seeing the tag select on the reading list page, and the desktop layout has gone funky (using a private browsing window)...

screenshot showing mobile layout with no tag filter

screenshot showing desktop view with content rendered in small space to the side

In Firefox it's fine though - maybe an issue with the media query in Chrome?

**Comment by nickytonline** [comment]
> In Chrome I'm not seeing the tag select on the reading list page, and the desktop layout has gone funky (using a private browsing window)...
> 
> screenshot showing mobile layout with no tag filter
> 
> screenshot showing desktop view with content rendered in small space to the side
> 
> In Firefox it's fine though - maybe an issue with the media query in Chrome?

Hmm, I'll take a peek as I've been working in Firefox lately. Seemed fine when running E2E tests in Chrome though. 🤔

**Comment by nickytonline** [comment]
@aitchiss, I fixed the issue in Chrome. Bad media queries on my end. It also removed an issue I thought I had with media queries in Cypress, so I no longer need to mock `matchMedia` 🆒

File: app/javascript/readingList/components/TagList.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+            onClick={onSelectTag}
+            key={tag}
+            className="opacity-0"
+            checked={selectedTag === tag}
+            value={tag}
+          />
+          #{tag}
+        </label>
+      ))}
+    </fieldset>
```
Comment (ludwiczakpawel) [inline app/javascript/readingList/components/TagList.jsx:27]: Using radio input here is a nice hack but it makes the nav accessibility a bit weird: when using my keyboard [tab] it focus the first element only and then jumps straight to articles list.. when going back [shift+tab] it highlights the last element only... Since this is radio, I'd have to use arrows to navigate through these radio options but that seems a bit confusing.

https://user-images.githubusercontent.com/108287/110937032-15117080-8332-11eb-9863-2208ecf2f68f.mov

(I added red outline to highlight focused elements in my vide but for some reason it didn't apply to tags :P but i hope you can spot what I'm talking about)

My suggestion would be replacing radio elements with classic links (or at least buttons?) but I can imagine it will require a bit more refactoring. But having form elements here instead of regular links doesn't seem to be the right choice.

**Review by ludwiczakpawel — CHANGES_REQUESTED** [review]
Couple things I've noticed:

1. There's that relayout jump on the page load - I think we should _always_ display sidebar even if there's no tags (which in most cases will happen only when there're no saved articles). 
    https://user-images.githubusercontent.com/108287/110935256-7c79f100-832f-11eb-86a4-8ea900d48f72.mov
2. On the left sidebar I should be able to "reset" my view - so I can see all saved articles again.. Probably having a link like "Everything" would be enough. (that's also missing in the mobile select)
    image
3. If there're no saved articles, we can probably hide select on mobile. To avoid "relayout jump", when there are articles, we can show this select on the same line as "View archive" button maybe?
4. This wide "View archive" button seems a bit off. Maybe instead of using `grid` in header (which causes the issue in the first place) you could use flexbox? 
    image

**Comment by ludwiczakpawel** [comment]
Oh, and that is still an issue:

> 3. When I use keyboard to choose a tag in this select, then hit Shift + Tab to go to previous field (search field), the filtering is being reset:

https://user-images.githubusercontent.com/108287/110938304-f44a1a80-8333-11eb-848a-bc06fe60c0df.mov

File: app/javascript/readingList/components/TagList.jsx (hunk 8 above)
Comment (ludwiczakpawel) [reply app/javascript/readingList/components/TagList.jsx:27]: Oh, I've just noticed your older comment about refactoring that in a separate PR. Sure. 

> In regards to keeping the filter in the URL, I completely agree, but that should be a separate PR a that changes how the entire state is managed for this component.

^ I agree but in this case I wouldn't refactor the nav to be a form radio inputs (they currently are links in prod), since we will circle back to this anyway and turn them to be links again - instead probably only make adjustments to the `onSelectTag` function so we can avoid that UX regression with keyboard navigation introduced by turning links into form elements.

Comment (aitchiss) [reply app/javascript/readingList/components/TagList.jsx:27]: I agree with @ludwiczakpawel I think for a sighted user it's not clear how to navigate the tags when they don't look like radios. 

Thinking about it, at the moment it's almost behaving more like a set of Tabs, with a single active page/tab at a time, but that also doesn't feel intuitive with the UI. It's been suggested already, but making these links and having the filter in the URL makes a lot of sense, and when we do that work this list can become a secondary `nav`.

tl;dr I second @ludwiczakpawel's suggestion to keep them as they are until we make that change.

File: app/javascript/readingList/components/TagList.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  );
+}
+
+/**
+ *
+ * @param {object} props
+ * @param {Array<string>} props.availableTags A list of available tags.
+ * @param {string} [props.selectedTag=''] The currently selected tag.
+ * @param {function} A handler for when the selected tag changes.
```
Comment (aitchiss) [inline app/javascript/readingList/components/TagList.jsx:36]:
```suggestion
 * @param {function} props.onSelectTag A handler for when the selected tag changes.
```
Also we should add the `isMobile` param

File: app/javascript/readingList/components/TagList.jsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+  onSelectTag,
+  isMobile = false,
+}) {
+  return isMobile ? (
+    <select
+      class="crayons-select"
+      aria-label="Filter by tag"
+      onBlur={(event) => {
+        // We need blur for a11y, but we also don't want to make the same search query twice
+        // if the tag hasn't changed since the previous onChange.
```
Comment (aitchiss) [inline app/javascript/readingList/components/TagList.jsx:50]: I don't think we do need it anymore - the linter rule for this is deprecated as the issues were reported on lower than IE 10 https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md

File: app/javascript/readingList/readingList.jsx [hunk 11]
```diff
@@ -186,17 +193,17 @@ export class ReadingList extends Component {
       ''
     );
     return (
-      <section>
-        <header className="crayons-layout flex justify-between items-center pb-0">
+      <main>
```
Comment (aitchiss) [inline app/javascript/readingList/readingList.jsx:196]: this is a small thing but it makes me very happy 😄

File: docs/tests/e2e-tests.md [hunk 12]
```diff
@@ -108,6 +108,42 @@ Pro as there were issues integrating the cypress-rails gem with the
 [knapsack-pro-cypress](https://github.com/KnapsackPro/knapsack-pro-cypress) npm
 package.
 
+## Custom Cypress commands
+
+[Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands.html)
+allow you do extend the functionality of the E2E testing framework. In the case
```
Comment (aitchiss) [inline docs/tests/e2e-tests.md:114]:
```suggestion
allow you to extend the functionality of the E2E testing framework. In the case
```

File: app/javascript/readingList/components/TagList.jsx (hunk 10 above)
Comment (nickytonline) [reply app/javascript/readingList/components/TagList.jsx:50]: I turned off this rule in #12986.

File: app/javascript/readingList/components/TagList.jsx (hunk 8 above)
Comment (nickytonline) [reply app/javascript/readingList/components/TagList.jsx:27]: I can put it back and disable the eslint rule that complains we're using a link like a button. Like already mentioned, the links in prod have click event listeners on them already.

**Comment by nickytonline** [comment]
> Couple things I've noticed:
> 
>     1. There's that relayout jump on the page load - I think we should _always_ display sidebar even if there's no tags (which in most cases will happen only when there're no saved articles).

Fixed. The sidebar is always displayed now in the wide view. It actually makes sense now that there is also a link to show all articles.

image

image


>     2. On the left sidebar I should be able to "reset" my view - so I can see all saved articles again.. Probably having a link like "Everything" would be enough. (that's also missing in the mobile select)

There is now the `All tags` option.

image

>     3. If there're no saved articles, we can probably hide select on mobile. To avoid "relayout jump", when there are articles, we can show this select on the same line as "View archive" button maybe?

Given that we have the All tags option now, the select would only have that option, or do you think it makes more sense to hide it still and also in the wide view to not render the `All tags` link?

>     4. This wide "View archive" button seems a bit off. Maybe instead of using `grid` in header (which causes the issue in the first place) you could use flexbox?

Taking a peek at that.

**Comment by nickytonline** [comment]
Alright, finally got back to this PR. This should be in a good state to check out now @ludwiczakpawel and @aitchiss.

File: app/javascript/readingList/__tests__/readingList.test.jsx [hunk 13]
```diff
@@ -62,16 +62,4 @@ describe('<ReadingList />', () => {
 
     expect(results).toHaveNoViolations();
   });
-
-  it('renders all the elements', () => {
```
Comment (nickytonline) [inline app/javascript/readingList/__tests__/readingList.test.jsx:66]: Covered by the E2E tests now.

File: app/javascript/readingList/readingList.jsx [hunk 14]
```diff
@@ -187,17 +193,17 @@ export class ReadingList extends Component {
...
+        <header className="crayons-layout l:grid-cols-2 pb-0">
           <h1 class="crayons-title">
             {isStatusViewValid ? 'Reading list' : 'Archive'}
             {` (${itemsTotal})`}
           </h1>
-
-          <div class="flex items-center">
+          <fieldset className="grid gap-2 m:flex m:justify-end l:mb-0 mb-2 px-2 m:px-0">
```
Comment (ludwiczakpawel) [inline app/javascript/readingList/readingList.jsx:202]: If you add `m:items-center` it will better display button and input (right now, it's kinda enlarged).

**Review by ludwiczakpawel — CHANGES_REQUESTED** [review]
Much better mate, thanks! It's almost good to go:)

There're still two issues:

1. Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit `Shift + Tab` again (to focus on the "View archive" button - it will reset the tags filtering.
2. Every time I click through the tags in left sidebar, I can see "empty state" for split of a second. Can we not have it while fetching the results? I don't even think we need to show a spinner since it happens almost instantly. 

Also, one tiny suggestion in code.

**Comment by nickytonline** [comment]
> Much better mate, thanks! It's almost good to go:)
> 
> 
> 
> There're still two issues:
> 
> 
> 
> 1. Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit `Shift + Tab` again (to focus on the "View archive" button - it will reset the tags filtering.
> 
> 2. Every time I click through the tags in left sidebar, I can see "empty state" for split of a second. Can we not have it while fetching the results? I don't even think we need to show a spinner since it happens almost instantly. 
> 
> 
> 
> Also, one tiny suggestion in code.

For 2., it was already like that. I was trying to keep the scope of the PR to just making things responsive, butI can probably squeak that in. 😎

**Comment by ludwiczakpawel** [comment]
> For 2., it was already like that. I was trying to keep the scope of the PR to just making things responsive, butI can probably squeak that in. 😎

You're right! Yea, if you find it being a quick fix would be nice to do it. If not, that's absolutely fine then :)

File: app/javascript/readingList/readingList.jsx (hunk 14 above)
Comment (nickytonline) [reply app/javascript/readingList/readingList.jsx:202]: Done.

**Comment by nickytonline** [comment]
> Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit Shift + Tab again (to focus on the "View archive" button - it will reset the tags filtering.

Fixed. This was a regression caused by moving to filtering by one tag only @ludwiczakpawel.

Going to look at fixing the empty state issue and some tests and this will be good to go for hopefully a final review. 😅

File: app/javascript/readingList/__tests__/readingList.test.jsx [hunk 15]
```diff
@@ -54,7 +65,7 @@ describe('<ReadingList />', () => {
     delete global.getCsrfToken;
   });
 
-  it('should have no a11y violations', async () => {
+  it('should have no a11y violations on small screens', async () => {
```
Comment (aitchiss) [inline app/javascript/readingList/__tests__/readingList.test.jsx:68]: nitpick - this test and the "should have no a11y violations on large screens" looks to be the exact same?

File: app/javascript/readingList/components/TagList.jsx [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+          }`}
+          data-no-instant
+          data-tag={tag}
+          onClick={onSelectTag}
+          key={tag}
+          href={`t/${tag}`}
+        >
+          #{tag}
+        </a>
+      ))}
```
Comment (aitchiss) [inline app/javascript/readingList/components/TagList.jsx:30]: refactor (blocking) - can we make these links list items inside a `<ul>`?

File: app/javascript/readingList/components/TagList.jsx [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+    </nav>
+  );
+}
+
+/**
+ *
+ * @param {object} props
+ * @param {Array<string>} props.availableTags A list of available tags.
+ * @param {string} [props.selectedTag=''] The currently selected tag.
+ * @param {function} props.onSelectTag A handler for when the selected tag changes.
```
Comment (aitchiss) [inline app/javascript/readingList/components/TagList.jsx:40]: nitpick (blocking) - we're missing the `isMobile` prop here

File: app/javascript/readingList/readingList.jsx [hunk 18]
```diff
@@ -187,17 +194,17 @@ export class ReadingList extends Component {
       ''
     );
     return (
-      <section>
-        <header className="crayons-layout flex justify-between items-center pb-0">
+      <main>
```
Comment (aitchiss) [inline app/javascript/readingList/readingList.jsx:197]: suggestion (not-blocking) - if we add the id `main-content` here, the skip link would work. Could also be picked up in a separate PR though as it's not really related to this issue

```suggestion
      <main id="main-content">
```

File: app/javascript/readingList/readingList.jsx [hunk 19]
```diff
@@ -206,35 +213,70 @@ export class ReadingList extends Component {
               {isStatusViewValid ? 'View archive' : 'View reading list'}
             </Button>
             <input
-              aria-label="Search..."
+              aria-label="Filter reading list by text"
```
Comment (aitchiss) [inline app/javascript/readingList/readingList.jsx:216]: so much better 😄

**Review by aitchiss — CHANGES_REQUESTED** [review]
Checked this locally on desktop and android mobile and all looking good - I just have some v minor change requests/suggestions 🙂

**Review by ludwiczakpawel — APPROVED** [review]
NICE!

File: app/javascript/readingList/__tests__/readingList.test.jsx (hunk 15 above)
Comment (nickytonline) [reply app/javascript/readingList/__tests__/readingList.test.jsx:68]: Oops, will fix that.

File: app/javascript/readingList/components/TagList.jsx (hunk 17 above)
Comment (nickytonline) [reply app/javascript/readingList/components/TagList.jsx:40]: done

File: app/javascript/readingList/components/TagList.jsx (hunk 16 above)
Comment (nickytonline) [reply app/javascript/readingList/components/TagList.jsx:30]: done

File: app/javascript/readingList/__tests__/readingList.test.jsx (hunk 15 above)
Comment (nickytonline) [reply app/javascript/readingList/__tests__/readingList.test.jsx:68]: done

**Comment by nickytonline** [comment]
I also tested this in iOS 14.4.2 and all looks good.

**Review by aitchiss — APPROVED** [review]
Nice! 🚀
